android/src/main/java/io/radar/react/RNRadarReceiver.java

```diff
@@ -26,14 +27,34 @@ public class RNRadarReceiver extends RadarReceiver {
     private ReactNativeHost reactNativeHost;
     private static final String TAG = "RNRadarReceiver";
 
+    private int listenerCount = 0;
+    boolean hasListeners = false;
+
     private void sendEvent(final String eventName, final Object data) {
         final ReactInstanceManager reactInstanceManager = reactNativeHost.getReactInstanceManager();
         ReactContext reactContext = reactInstanceManager.getCurrentReactContext();
-        if (reactContext != null) {
+        if (reactContext != null && hasListeners) {
             reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit(eventName, data);
         }
     }
 
+    @ReactMethod
+    public void addListener(String eventName) {
```

Comment:
Can you provide more context on the adding and removing listeners and how it controls whether events are emitted or not?

Reply:
these are for emitting events from native modules to javascript code. if no event listeners are registered, then we don't emit these events. we get compilation warnings when we dont implement add/remove listener functions
https://reactnative.dev/docs/native-modules-android#sending-events-to-javascript